src/request.rs

```diff
@@ -445,6 +451,24 @@ impl Request for Rename {
     const METHOD: &'static str = "textDocument/rename";
 }
 
+#[derive(Debug)]
+pub enum DocumentColor {}
+
+impl Request for DocumentColor {
+    type Params = DocumentColorParams;
+    type Result = Option<Vec<ColorInformation>>;
```

Comment:
The spec seems to specify `ColorPresentation[]` as the return value so we can remove the `Option`.